api/apps/api/test/access-control/project-acl.fixtures.ts

```diff
@@ -21,37 +22,95 @@ export const getFixtures = async () => {
   const projectViewerRole = Roles.project_viewer;
   const projectContributorRole = Roles.project_contributor;
   const projectOwnerRole = Roles.project_owner;
+  let projectCreatedId = '';
   const userProjectsRepo: Repository<UsersProjectsApiEntity> = app.get(
     getRepositoryToken(UsersProjectsApiEntity),
   );
   const cleanups: (() => Promise<void>)[] = [];
 
   return {
     cleanup: async () => {
-      await userProjectsRepo.delete({
-        userId: In([
-          ownerUserId,
-          contributorUserId,
-          viewerUserId,
-          otherOwnerUserId,
-        ]),
-      });
-      await Promise.all(cleanups.map((clean) => clean()));
+      for (const cleanup of cleanups.reverse()) {
+        await cleanup();
+      }
       await app.close();
     },
 
+    userRoleCleanup: async () => {
+      if (projectCreatedId) {
+        cleanups.push(
+          async () =>
+            await ProjectsACLTestUtils.deleteUserFromProject(
+              app,
+              ownerUserToken,
+              projectCreatedId,
+              otherOwnerUserId,
+            ),
+        );
+        cleanups.push(
+          async () =>
+            await ProjectsACLTestUtils.deleteUserFromProject(
+              app,
+              ownerUserToken,
+              projectCreatedId,
+              viewerUserId,
+            ),
+        );
+        cleanups.push(
+          async () =>
+            await ProjectsACLTestUtils.deleteUserFromProject(
+              app,
+              ownerUserToken,
+              projectCreatedId,
+              contributorUserId,
+            ),
+        );
+      }
+    },
+
+    GivenProjectExistsAndHasUsers: async () => {
+      const result = await userProjectsRepo.findOne({
+        where: {
+          userId: ownerUserId,
+          roleName: projectOwnerRole,
+        },
+      });
+
+      if (!result) {
+        throw new Error('Project does not exist');
+      }
+
+      return result.projectId;
+    },
+
     GivenProjectWasCreated: async () => {
       const { cleanup, projectId } = await GivenProjectExists(
         app,
         ownerUserToken,
       );
       cleanups.push(cleanup);
+      projectCreatedId = projectId;
```

Comment:
Second thought: we can extract it as a `background`  (see https://cucumber.io/docs/gherkin/reference/) and just put project creation after boostrapping the application